src/modules/ekf2/EKF2.cpp

```diff
@@ -1430,25 +1406,22 @@ void EKF2::UpdateMagCalibration(const hrt_abstime &timestamp)
 			// Declare all bias estimates invalid if any variances are out of range
 			bool all_estimates_invalid = false;
 
-			float covariances[24];
-			_ekf.covariances_diagonal().copyTo(covariances);
+			const Vector3f mag_bias_variance{_ekf.getMagBiasVariance()};
 
 			for (uint8_t axis_index = 0; axis_index <= 2; axis_index++) {
-				if (covariances[axis_index + 19] < min_var_allowed
-				    || covariances[axis_index + 19] > max_var_allowed) {
+				if (mag_bias_variance(axis_index) < min_var_allowed
+				    || mag_bias_variance(axis_index) > max_var_allowed) {
 					all_estimates_invalid = true;
 				}
 			}
```

Comment:
This would be a bit more compact.
```
const bool all_estimates_invalid = mag_bias_variance.min() < min_var_allowed &&  mag_bias_variance.max() > max_var_allowed;
```

Reply:
Done.